src/honeybadger.cr

```diff
@@ -3,30 +3,65 @@ require "./honeybadger/*"
 module Honeybadger
   VERSION = "0.1.0"
 
-  @@api_key = ""
-  @@report_data = true
+  class Configuration
+    # A Honeybadger API key.
+    property api_key = ""
+
+    # The API endpoint for sending Honeybadger payloads.
+    property endpoint : Path = Path["https://api.honeybadger.io"]
+
+    # The project git revision. Evaluated at compile time.
+    getter git_revision : String = {{ run("./run_macros/git_revision.cr").stringify }}.strip
+
+    # The system or container hostname.
+    property hostname : String = System.hostname
+
+    # The path to the projects source code. Evaluated at compile time.
+    property project_root : String = {{ run("./run_macros/pwd.cr").stringify }}.strip
+
+    # Send data to the Honeybadger collector
+    property report_data = true
+  end
+
+  # The Honeybadger Configuration singleton instance.
+  class_getter configuration = Configuration.new
 
   # Set the API key to be used for calls to the honeybadger API.
-  # Call `configure` during application pre-boot:
+  # Call `configure` during an application boot sequence and populate the api key:
   #
   # ```
   #   honeybadger_api_key = ENV["HONEYBADGER_API_KEY"]? || "00000000"
-  #   honeybadger_enabled = true
-  #
   #   Honeybadger.configure(api_key: honeybadger_api_key)
   # ```
-  def self.configure(api_key : String, report_data = true) : Nil
-    @@api_key = api_key
-    @@report_data = report_data
+  def self.configure(api_key : String, *, report_data = true) : Nil
```

Comment:
I specifically left report_data here because it seems like it's the most likely to be used in a "lightweight" configuration mode, especially until #9 is implemented.